docs/viewlifecycle.md

```diff
@@ -0,0 +1,439 @@
+# View Lifecycle
+
+The Marionette views use an event lifecycle, triggering events on any listeners
+to act at different points inside the creation and destruction of views and
+their children.
+
+## Documentation Index
+
+* [`View` Lifecycle](#view-lifecycle)
+  * [View Creation Lifecycle](#view-creation-lifecycle)
+  * [View Destruction Lifecycle](#view-destruction-lifecycle)
+  * [View Creation Events](#view-creation-events)
+  * [View Destruction Events](#view-destruction-events-events)
+  * [Other View Events](#other-view-events)
+* [`CollectionView` Lifecycle](#collectionview-lifecycle)
+  * [Collection Creation Lifecycle](#collection-creation-lifecycle)
+  * [Collection Destruction Lifecycle](#collection-destruction-lifecycle)
+  * [Collection Creation Events](#collection-creation-events)
+  * [Collection Destruction Events](#collection-destruction-events-events)
+  * [Other Collection Events](#other-collection-events)
+
+## `View` Lifecycle
+Marionette views defined a number of events during the creation and destruction
+lifecycle - when the view is displayed in and emptied from a region. In the
+documentation, we will reference the event name, though `onEvent` handling can
+be used.
+
+### View Creation Lifecycle
+
+When a view is initialized and then displayed inside a region (using
+`showChildView()`) a set of events will be called in a specific order.
+
+| Order |      Event      |
+| :---: |-----------------|
+|   1   | `before:render` |
+|   2   | `render`        |
+|   3   | `before:attach` |
+|   4   | `attach`        |
+|   5   | `dom:refresh`   |
+
+### View Destruction Lifecycle
+
+When  `region.empty()` is called, the view will be destroyed, calling events as
+part of the destruction lifecycle.
+
+| Order |       Event       |
+| :---: |-------------------|
+|   1   |  `before:destroy` |
+|   2   |  `before:detach`  |
+|   3   |  `detach`         |
+|   4   |  `destroy`        |
+
+### View Creation Events
+
+These events are fired during the view's creation and rendering in a region.
+
+#### View `before:render`
+
+Triggered before a View is rendered.
+
+```javascript
+var Mn = require('backbone.marionette');
+
+Mn.View.extend({
+  onBeforeRender: function() {
+    // set up final bits just before rendering the view's `el`
+  }
+});
+```
+
+#### View `render`
+
+Triggered after the view has been rendered.
+You can implement this in your view to provide custom code for dealing
+with the view's `el` after it has been rendered.
+
+```javascript
+var Mn = require('backbone.marionette');
+
+Mn.View.extend({
+  onRender: function() {
+    console.log('el exists but is not visible in the DOM');
+  }
+});
+```
+
+#### View `before:attach`
+
+Triggered after the View has been rendered but just before it is first bound
+into the page DOM. This will only be triggered once per `region.show()` - if
+you want something that will be triggered every time the DOM changes,
```

Comment:
Maybe don't include
>  - if you want something that will be triggered every time the DOM changes,  you may want `render` or `before:render`.

This leads me to think that if I do
`this.ui.myEl.append('<span></span>')` then the render event will trigger, when that's not the case.